junit-jupiter-params/src/main/java/org/junit/jupiter/params/support/AnnotationConsumerInitializer.java

```diff
@@ -47,14 +52,28 @@ private AnnotationConsumerInitializer() {
 	public static <T> T initialize(AnnotatedElement annotatedElement, T annotationConsumerInstance) {
 		if (annotationConsumerInstance instanceof AnnotationConsumer) {
 			Class<? extends Annotation> annotationType = findConsumedAnnotationType(annotationConsumerInstance);
-			Annotation annotation = AnnotationUtils.findAnnotation(annotatedElement, annotationType) //
-					.orElseThrow(() -> new JUnitException(annotationConsumerInstance.getClass().getName()
-							+ " must be used with an annotation of type " + annotationType.getName()));
-			initializeAnnotationConsumer((AnnotationConsumer) annotationConsumerInstance, annotation);
+			List<? extends Annotation> annotations = streamAnnotations(annotatedElement, annotationType);
+
+			if (annotations.isEmpty()) {
+				throw new JUnitException(annotationConsumerInstance.getClass().getName()
+						+ " must be used with an annotation of type " + annotationType.getName());
+			}
+
+			annotations.forEach(annotation -> initializeAnnotationConsumer(
+				(AnnotationConsumer) annotationConsumerInstance, annotation));
 		}
 		return annotationConsumerInstance;
 	}
 
+	private static List<? extends Annotation> streamAnnotations(AnnotatedElement annotatedElement,
+			Class<? extends Annotation> annotationType) {
+
+		return annotationType.isAnnotationPresent(Repeatable.class)
+				? findRepeatableAnnotations(annotatedElement, annotationType)
+				: Stream.of(findAnnotation(annotatedElement, annotationType)).filter(Optional::isPresent).map(
+					Optional::get).collect(Collectors.toList());
```

Comment:
```suggestion
				: findAnnotation(annotatedElement, annotationType).map(Collections::singletonList).orElse(Collections.emptyList());
```

Reply:
Agree, much better than streaming the Optional 👍  
I've also made use of static imports for `Collections.emptyList()`, I hope this is fine.